pkg/scheduler/apis/config/validation/validation.go

```diff
@@ -72,6 +77,45 @@ func ValidateKubeSchedulerLeaderElectionConfiguration(cc *config.KubeSchedulerLe
 	return allErrs
 }
 
+func validateTopologySpreadConstraints(constraints []v1.TopologySpreadConstraint, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	for i, c := range constraints {
+		p := fldPath.Index(i)
+		if c.MaxSkew <= 0 {
+			f := p.Child("maxSkew")
+			allErrs = append(allErrs, field.Invalid(f, c.MaxSkew, "must be greater than zero"))
+		}
+		if len(c.TopologyKey) == 0 {
+			f := p.Child("topologyKey")
+			allErrs = append(allErrs, field.Required(f, "can not be empty"))
+		}
+		if !supportedScheduleActions.Has(string(c.WhenUnsatisfiable)) {
+			f := p.Child("whenUnsatisfiable")
+			allErrs = append(allErrs, field.NotSupported(f, c.WhenUnsatisfiable, supportedScheduleActions.List()))
+		}
+		if c.LabelSelector != nil {
+			f := field.Forbidden(p.Child("labelSelector"), "cluster-level constraint must not define a selector")
```

Comment:
Probably explain more on why we disallow specifying labelSelector here.

Reply:
Done